src/runtime/HalideBuffer.h

```diff
@@ -371,7 +371,7 @@ class Buffer {
     void crop_host(int d, int min, int extent) {
         assert(dim(d).min() <= min);
         assert(dim(d).max() >= min + extent - 1);
-        int shift = min - dim(d).min();
+        ptrdiff_t shift = min - dim(d).min();
         if (buf.host != nullptr) {
             buf.host += shift * dim(d).stride() * type().bytes();
```

Comment:
I don't think this is necessary since `shift` (now a `ptrdiff_t`) should force promotion of the other field in the expression.  Is that not correct?

Reply:
I was thinking it would depend on the order in which the compiler associates this arithmetic. Even if it's well defined behavior for the compiler to implement it as (shift * dim(d).stride()) * type().bytes(), we should at least add parentheses to make that explicit.

Reply:
I believe the associativity is guaranteed to be left-to-right: https://en.cppreference.com/w/c/language/operator_precedence

Reply:
It may be, but I still think we should not rely on this, by adding explicit parentheses. It's easy for things like this to unintentionally break later because it is so subtle.

Reply:
OK. And BTW, thanks for reviewing this.
I can get back to modifying this PR tomorrow afternoon and add explicit parenthesis, though frankly I think it actually makes things more confusing.  I would look at such code and do a double take, wondering why those parenthesis are there at all, thinking I was actually missing something subtle.  The same applies to unnecessary casts.  IMO, if your goal is to inform developers who run across this code later, it's better to just add appropriate comments.  If for some reason you don't trust the compiler to follow the C++ spec, then the correct approach is to add unit tests to actually verify behavior, which can then also detect regressions.
These are also the sorts of problems that can often be detected by static analysis.

Reply:
I'm not worried about a non-conforming compiler, I'm more thinking that someone might come along and refactor the code somehow (e.g. pulling something into a temporary, reordering the expression) without realizing the order is important.

Reply:
Added parenthesis